client/interface.go

```diff
@@ -58,7 +58,7 @@ type APIClient interface {
 	ImageSave(imageIDs []string) (io.ReadCloser, error)
 	ImageTag(options types.ImageTagOptions) error
 	Info() (types.Info, error)
-	NetworkConnect(networkID, containerID string, config *network.NetworkingConfig) error
+	NetworkConnect(networkID, containerID string, config *network.EndpointSettings) error
```

Comment:
If we are never going to support connecting to multiple networks on a single call, it is fine to remove this.